nodejs/src/index.js

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright 2016, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+'use strict';
+
+const googleAuth = require('google-auto-auth');
+const got = require('got');
+const path = require('path');
+const snakeCase = require('lodash.snakecase');
+
+const PAGE_SIZE = 100;
+
+function fetchPage (url, authToken, nextPageToken) {
+  const query = {
+    pageSize: PAGE_SIZE,
+    returnValues: true
+  };
+
+  if (nextPageToken) {
+    query.pageToken = nextPageToken;
+  }
+
+  return got.get(url, {
+    headers: {
+      'Authorization': `Bearer ${authToken}`
+    },
+    query,
+    json: true
+  }).then((response) => {
+    const variables = response.body.variables || [];
+    if (variables.length < PAGE_SIZE) {
+      return variables;
+    } else {
+      return fetchPage(url, authToken, response.body.nextPageToken)
+        .then((_variables) => variables.concat(_variables || []));
+    }
+  });
+}
+
+/**
+ * Retrieves all variables in the given config.
+ *
+ * @param {string} configName
+ * @param {object} [opts]
+ * @returns {Promise}
+ */
+exports.getVariables = (configName, opts = {}) => {
+  opts.projectId || (opts.projectId = process.env.GCLOUD_PROJECT || process.env.GOOGLE_CLOUD_PROJECT);
```

Comment:
I think the Python and Go implementations have this flipped (prefer GOOGLE_CLOUD_PROJECT var over GCLOUD_PROJECT)